runtime/compiler/trj9/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -1965,26 +1965,26 @@ J9::Z::TreeEvaluator::asynccheckEvaluator(TR::Node * node, TR::CodeGenerator * c
                updateReferenceNode(firstChild, src1Reg);
                firstChild->setRegister(src1Reg);
 
-               generateRIInstruction(cg, TR::InstOpCode::getCmpHalfWordImmOpCodeFromNode(firstChild), node, src1Reg, value);
+               generateRIInstruction(cg, TR::InstOpCode::getCmpHalfWordImmOpCode(), node, src1Reg, value);
```

Comment:
For the `asynccheckEvaluator` this change is safe because of the way `TR::asynccheck` is lowered in OpenJ9, which can be found here:

https://github.com/eclipse/openj9/blob/3d433849504e966ad88e850ba5af80eab35f6bb7/runtime/compiler/trj9/env/VMJ9.cpp#L3461-L3487

on line 3471 in the above snippet we create the first child of the `TR::asynccheck` check using the `udataLoadOpCode` API which under the hood looks like:

https://github.com/eclipse/openj9/blob/3d433849504e966ad88e850ba5af80eab35f6bb7/runtime/compiler/trj9/env/VMJ9.cpp#L3437-L3447

Hence `getCmpHalfWordImmOpCodeFromNode` is equivalent to `getCmpHalfWordImmOpCode`.

Reply:
I wonder if we can delete this entire block.  Given the dependency on `lowerAsyncCheck`, I cannot see how one can ever get a MemRef for the first child where Index / Base Reg are both non-NULL; maybe if offset requires long displacement .. but then, I'd argue we'd have a perf issue too.

Reply:
> I wonder if we can delete this entire block. Given the dependency on lowerAsyncCheck, I cannot see how one can ever get a MemRef for the first child where Index / Base Reg are both non-NULL; maybe if offset requires long displacement .. but then, I'd argue we'd have a perf issue too.

I agree. Did not notice this myself. I'll open up an item to get this fixed on our backlog.